### ISSUE TITLE
Catch attempts to post task with invalid or unsupported tool.

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#1125]
`POST`s to the API's `/tasks` endpoint requesting a tool that exists but can't run the task causes the database's `api_task_post()` to throw an exception.  This condition should be caught in the API during the tool selection process.

```
pscheduler-api ERROR    Exception: psycopg2.InternalError: Tool "bwctltraceroute" is not available
pscheduler-api ERROR    Traceback (most recent call last):
pscheduler-api ERROR      File "/usr/lib/python3.6/site-packages/flask/app.py", line 1950, in full_dispatch_request
pscheduler-api ERROR        rv = self.dispatch_request()
pscheduler-api ERROR      File "/usr/lib/python3.6/site-packages/flask/app.py", line 1936, in dispatch_request
pscheduler-api ERROR        return self.view_functions[rule.endpoint](**req.view_args)
pscheduler-api ERROR      File "/var/www/pscheduler-server/pschedulerapiserver/tasks.py", line 477, in tasks
pscheduler-api ERROR        task.get("priority", None), diags], onerow=True)
pscheduler-api ERROR      File "/var/www/pscheduler-server/pschedulerapiserver/dbcursor.py", line 112, in dbcursor_query
pscheduler-api ERROR        raise ex
pscheduler-api ERROR      File "/var/www/pscheduler-server/pschedulerapiserver/dbcursor.py", line 97, in dbcursor_query
pscheduler-api ERROR        cursor.execute(query, args)
pscheduler-api ERROR    psycopg2.InternalError: Tool "bwctltraceroute" is not available
```

The API should kick a bad request back to the caller rather than a 500 since this isn't an internal problem.
